packages/database/src/core/view/filter/LimitedFilter.ts

```diff
@@ -119,19 +125,17 @@ export class LimitedFilter implements NodeFilter {
         let count = 0;
         while (iterator.hasNext() && count < this.limit_) {
           const next = iterator.getNext();
-          let inRange;
-          if (this.reverse_) {
-            inRange =
-              this.index_.compare(this.rangedFilter_.getStartPost(), next) <= 0;
-          } else {
-            inRange =
-              this.index_.compare(next, this.rangedFilter_.getEndPost()) <= 0;
-          }
+          const inRange =
+            this.withinDirectionalStart(next) &&
+            this.withinDirectionalEnd(next);
           if (inRange) {
             filtered = filtered.updateImmediateChild(next.name, next.node);
             count++;
+          } else if (!this.withinDirectionalStart(next)) {
```

Comment:
More of a question than anything else. Why did this need to get added? And it looks like we're adding an additional check on line 130

Reply:
I think this could be simplified to:
```
        while (iterator.hasNext() && count < this.limit_) {
          const next = iterator.getNext();
          if (!this.withinDirectionalStart(next)) {
            continue;
          }
          if (!this.withinDirectionalEnd(next)) {
            break;
          }
          filtered = filtered.updateImmediateChild(next.name, next.node);
          count++;
        }
```

Currently, it's mimicing [LimitedRangeView](https://github.com/firebase/firebase-server/blob/master/firebase-server/firebase-core/src/main/scala/com/firebase/core/namespace/view/LimitedRangeView.scala#L305-L312) on the backend. 

We considered teaching [`iteratorFrom`](https://github.com/firebase/firebase-server/blob/42585d4169124e76582bd07a7bf06e91bfaada5f/firebase-server/firebase-util/src/main/scala/com/firebase/util/data/Query.scala#L142) and `reverseIteratorFrom` about inclusive/exclusive bounds, but there are a few different implementations on the backend and it seemed like more trouble than it's worth.

Reply:
Done.